File: 백준/Silver/21921. 블로그/블로그.java
import java.io.*;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuffer sb = new StringBuffer();

        // n,x를 입력받음
        String[] input = br.readLine().split(" ");
        Integer n = Integer.parseInt(input[0]);
        Integer x = Integer.parseInt(input[1]);

        // arr를 입력받음
        String[] arr = br.readLine().split(" ");

        // Stream을 사용하여 초기 max값 설정
        Integer maxNum = Arrays.stream(Arrays.copyOfRange(arr,0,x))
                .mapToInt(Integer::parseInt)
                .sum();

        // 반복문을 사용하여 최댓값과 그 개수를 구함
        Integer cnt = 1;
        Integer before = maxNum;
        for (int i = 0; i < n - x; i++) {
            Integer left = Integer.parseInt(arr[i]);
            Integer right = Integer.parseInt(arr[i + x]);
            Integer cur = before - left + right;

            if (cur > maxNum) {
                cnt = 1;
                maxNum = cur;
            } else if (cur.equals(maxNum)) {
                cnt ++;
            }
            before = cur;
        }

        if (maxNum != 0) {
            sb.append(maxNum + "\n");
            sb.append(cnt);
        } else {
            sb.append("SAD");
        }
        System.out.println(sb);

    }

}

File: 백준/Silver/21921. 블로그/블로그.py
'''
Approach
    - 최대 방문자 수와 기간을 구하여라
    - 최대 방문자가 0이라면 'SAD'출력

>>> 투포인터
    : l = 0, r = l+x로 잡고 r이 n-1에 도착할때까지 진행하면서 최댓값을 비교해서 cnt한다
'''
from sys import stdin
input = stdin.readline

n,x = map(int,input().split())
visitor = list(map(int,input().split()))
maxNum = sum(visitor[:x])
cur = maxNum
cnt = 1
l = 0
r = l+x
while r < n:
    cur -= visitor[l]
    cur += visitor[r]
    
    if cur > maxNum:
        cnt = 1
        maxNum = cur
    elif cur == maxNum:
        cnt +=1
    
    l +=1
    r = l+x

if maxNum:
    print(maxNum)
    print(cnt)
else:
    print('SAD')